modules/highgui/CMakeLists.txt

```diff
@@ -68,6 +68,18 @@ file(GLOB highgui_ext_hdrs
 # Removing WinRT API headers by default
 list(REMOVE_ITEM highgui_ext_hdrs "${CMAKE_CURRENT_LIST_DIR}/include/opencv2/${name}/highgui_winrt.hpp")
 
+if(HAVE_GLFW)
+  add_definitions(-DHAVE_GLFW)
+  include_directories("${glfw3_DIR}/../../../include")
+  link_directories("${glfw3_DIR}/../../../lib")
```

Comment:
I propose to make the options as private as possible:
- use `add_target_definitions(${the_module} PRIVATE '-DHAVE_GLFW')`
- GLFW defines target with all required options. I propose just `target_link_libraries(${the_module} PRIVATE glfw)`. Link directories and include directories are not required.